cli/cmd/uninstall/uninstall.go

```diff
@@ -0,0 +1,509 @@
+		c.UI.Output("Name: %s", foundReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace %s", foundReleaseNamespace, terminal.WithInfoStyle())
+	}
+	// Prompt with a warning for approval before deleting PVCs, Secrets, Service Accounts, Roles, and Role Bindings.
+	if !c.flagAutoApprove {
+		confirmation, err := c.UI.Input(&terminal.Input{
+			Prompt: fmt.Sprintf("WARNING: Proceed with deleting PVCs, Secrets, Service Accounts, Roles, and Role Bindings for the following installation? \n\n   Name: %s \n   Namespace: %s \n\n   Only approve if all data from this installation can be deleted. (y/N)", foundReleaseName, foundReleaseNamespace),
+			Style:  terminal.WarningStyle,
+			Secret: false,
+		})
+		if err != nil {
+			c.UI.Output(err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+		if common.Abort(confirmation) {
+			c.UI.Output("Uninstall aborted without deleting PVCs and Secrets.", terminal.WithInfoStyle())
+			return 1
+		}
+	}
+
+	if err := c.deletePVCs(foundReleaseName, foundReleaseNamespace); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if err := c.deleteSecrets(foundReleaseName, foundReleaseNamespace); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if err := c.deleteServiceAccounts(foundReleaseName, foundReleaseNamespace); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if err := c.deleteRoles(foundReleaseName, foundReleaseNamespace); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if err := c.deleteRoleBindings(foundReleaseName, foundReleaseNamespace); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	return 0
+}
+
+func (c *Command) Help() string {
+	c.once.Do(c.init)
+	s := "Usage: consul-k8s uninstall [flags]" + "\n" + "Uninstall Consul with options to delete data and resources associated with Consul installation." + "\n\n" + c.help
+	return s
+}
+
+func (c *Command) Synopsis() string {
+	return "Uninstall Consul deployment."
+}
+
+func (c *Command) findExistingInstallation(actionConfig *action.Configuration) (bool, string, string, error) {
+	lister := action.NewList(actionConfig)
+	// lister.All will search for helm installations in all states, such as deployed, pending, uninstalling, etc.
+	lister.All = true
+	if c.flagNamespace == defaultAllNamespaces {
+		lister.AllNamespaces = true
+	}
+	res, err := lister.Run()
+	if err != nil {
+		return false, "", "", fmt.Errorf("error finding existing installations: %s", err)
+	}
+
+	found := false
+	foundReleaseName := ""
+	foundReleaseNamespace := ""
+	for _, rel := range res {
+		if rel.Chart.Metadata.Name == "consul" {
+			if c.flagNamespace != defaultAllNamespaces {
+				// If we found a chart named "consul" and -namespace was specified, we only found the release if the
+				// release namespace matches the -namespace flag.
+				if c.flagNamespace == rel.Namespace {
```

Comment:
I like the idea to combine it for readability! I think it's possible for empty string value for `rel.Namespace` to mean default namespace, and we can't test our way around that since we can't test the call to List itself, so it makes sense to combine the check